contracts/cw-blob/src/lib.rs

```diff
@@ -58,37 +58,38 @@ pub mod interface {
                 .checksum()
                 .unwrap()
         }
+    }
 
+    pub trait MigrationFromBlob<Chain: CwEnv>:
+        ContractInstance<Chain> + CwOrchUpload<Chain> + MigratableContract
+    {
         /// Instantiate blob and migrate to your desired contract.
         /// It will upload your contract, if it's not uploaded already
         ///
         /// Checksum of the uploaded blob_code_id on chain should match [CwBlob::checksum()]
-        pub fn upload_and_migrate<M>(
-            chain: T,
+        fn deterministic_instantiate(
```

Comment:
Is there a method like `deterministic_address` that allows a user to get the address that the contract will be deployed on? 

Should be useful.